docs/decisions/0004-axios-caching-implementation.rst

```diff
@@ -0,0 +1,87 @@
+
+Having a frontend caching layer would also allow for a stale version of a page to be displayed for a short amount of time in the case of a backend outage.
+
+When Frontend Caching Should Be Used
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The frontend cache should be used to eliminate the need to reload the same data on each page load as a user navigates around. It shouldn't be used to reduce the number of times multiple components on the same page try to make the same network call. Reducing the number of times the same network call is made on a single page should be done by moving the call higher up in the hierarchy so the call only gets made once and all lower level components are able to access the response data from it.
+
+Requests that are good candidates to use frontend caching are ones that:
+ * The data being returned changes very infrequently independently from the actions of the user on the page
+
+   - For example changes made by a background job or an admin making changes
+
+ * The data being returned isn't specific to the authenticated user
+
+ * The data returned is specific to the authenticated user but front end does not make the request when there isn't an authenticated user
+
+It should not be used as the only caching strategy that gets used. It is preferable to use a serverside caching strategy in conjunction with frontend caching.
+
+How To Use Frontend Caching
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The frontend caching layer can be used by passing {useCache: true} into the function that gets the httpClient.
+
+e.g.
+    service.getAuthenticatedHttpClient({ useCache: true });
+
+By default the response for a GET request will be stored in IndexedDB for 5 minutes and that cached value will get invalidated on any POST, PUT, PATCH, or DELETE request made to the same url. The caching layer also works with the standard cache-control headers: max-age, no-cache, and no-store
+
+Each request can also override the default cache configurations. `The How To document about caching <https://github.com/edx/frontend-platform/blob/master/docs/how_tos/caching.rst>`_ has more detailed examples about how this caching layer can be used.
+
+
+Implementation Details
+~~~~~~~~~~~~~~~~~~~~~~
+
+For both versions of the current AxiosJwtAuthService HTTP clients, the authenticated client and the base unauthenticated clients, there are now two additional versions that use the cache (i.e. CachedAuthenticated and CachedUnauthenticated). Bringing the total number of httpClient instances up to four.
+In order to keep the interface more simple the cached clients don’t have their own getter functions exposing this detail, in order to use the cached http client micro frontends pass in a param for using the cache with the current getter functions.
+
+In the case of the cached clients failing to get created the uncached clients will be returned and used instead.
```

Comment:
"In the case of failure during creation of cached client objects, an uncached client object will be returned and used instead."